src/rules/store/on-function-explicit-return-type.ts

```diff
@@ -16,22 +20,54 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
       category: 'Possible Errors',
       description: 'On function should have an explicit return type.',
       recommended: 'warn',
+      suggestion: true,
     },
     schema: [],
     messages: {
-      [messageId]:
+      [onFunctionExplicitReturnType]:
         'On functions should have an explicit return type when using arrow functions: `on(action, (state): State => {}`.',
+      [onFunctionExplicitReturnTypeSuggest]: 'Add explicit return type.',
     },
   },
   defaultOptions: [],
   create: (context) => {
+    const sourceCode = context.getSourceCode()
+
     return {
       [onFunctionWithoutType](node: TSESTree.ArrowFunctionExpression) {
         context.report({
           node,
-          messageId,
+          messageId: onFunctionExplicitReturnType,
+          suggest: [
+            {
+              fix: (fixer) => getFixes(node, sourceCode, fixer),
+              messageId: onFunctionExplicitReturnTypeSuggest,
+            },
+          ],
         })
       },
     }
   },
 })
+
+function getFixes(
+  { params }: TSESTree.ArrowFunctionExpression,
+  sourceCode: Readonly<TSESLint.SourceCode>,
+  fixer: TSESLint.RuleFixer,
+): TSESLint.RuleFix | readonly TSESLint.RuleFix[] {
+  const [firstParam] = params
+  const [lastParam] = params.slice(-1)
+  const previousToken = sourceCode.getTokenBefore(firstParam)
+  const isParenthesized =
+    previousToken && ASTUtils.isOpeningParenToken(previousToken)
+
+  if (isParenthesized) {
```

Comment:
I tried to use [`ASTUtils.isParenthesized`](https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/experimental-utils/src/ast-utils/eslint-utils/astUtilities.ts#L113-L116) but it doesn't work for this case, as it only returns `true` if a node has `parenthesis` before **and** after, which isn't necessarily our case, as we can have, for example, `on(action, (a, b) => ...`, so neither param `a`, nor param `b` would receive `true` as a response from this `function`.